Delete function rework

SpencerMartel opened the pull request:
Delete function now works with reworked wishlist_json as it became a dict, delete function now takes in any game in "x"

SpencerMartel committed:
Delete function rework

Delete function now works with reworked wishlist_json as it became a dict, delete function now takes in any game in "x"

Nwitz commented on steam.py:
```diff
@@ -125,17 +125,11 @@ def sync_wishlist_file():
+    app_to_delete = f'{app_tuple[0]}'
+    del wishlist_json[app_to_delete]
+    sync_wishlist_file()
+    reply = 'Game was successfully deleted, it is no longer being tracked'
```
Steam file shouldn't be responsible for knowing what a "reply" is. Instead it should tell the caller the result of the deletion, and let the caller decide how to proceed. 

In this case the steam file really only knows about managing local wishlist and performing fetches on the Steam API. Client.py should always be responsible for forming replies. 
This type of structure would allow us to call the delete function from other places that might not know what to do with a String reply.

SpencerMartel replied:
Gotcha, will make a class to pass back to steam.py to build the reply in there

Nwitz replied:
You could probably use a boolean (true, false) here since it either did delete, or didn't exist. I don't think we really have a case where the actual delete fails

Nwitz commented on steam.py:
```diff
@@ -125,17 +125,11 @@ def sync_wishlist_file():
+    del wishlist_json[app_to_delete]
+    sync_wishlist_file()
+    reply = 'Game was successfully deleted, it is no longer being tracked'
+    return wishlist_json, reply
```
Why are we returning the wishlist_json here?

SpencerMartel replied:
¯\_(ツ)_/¯
Will delete it

Nwitz commented on client.py:
```diff
@@ -36,8 +37,8 @@ async def on_message(message):
-    elif 'delete_' in user_input:
-        game_in_input = re.split('_',user_input)
+    elif 'delete' in user_input:
+        game_in_input = re.split('"',user_input)
```
Nice

SpencerMartel committed:
Delete_games + handle_delete_games functions fixed and implemented

Client side I added a function to handle deleting games and outputting results to discord. Fixed delete_function steam side from Noah's comments.

SpencerMartel committed:
Lil list_games & handle_list games changes

updated handle function for bettter clarity of what does what

SpencerMartel committed:
Stupid lil change

make discord users happy
